token/program-2022/src/pod.rs

```diff
@@ -0,0 +1,196 @@
+//! Rewrites of the base state types represented as Pods
+
+#[cfg(test)]
+use {
+    crate::state::{Account, Mint, Multisig},
+    solana_program::program_option::COption,
+};
+use {
+    crate::{instruction::MAX_SIGNERS, state::PackedSizeOf},
+    bytemuck::{Pod, Zeroable},
+    solana_program::{program_pack::IsInitialized, pubkey::Pubkey},
+    spl_pod::{
+        bytemuck::pod_get_packed_len,
+        primitives::{PodBool, PodU64},
+    },
+};
+
+/// [Mint] data stored as a Pod type
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+pub struct PodMint {
+    /// Optional authority used to mint new tokens. The mint authority may only
+    /// be provided during mint creation. If no mint authority is present
+    /// then the mint has a fixed supply and no further tokens may be
+    /// minted.
+    pub mint_authority: PodCOption<Pubkey>,
+    /// Total supply of tokens.
+    pub supply: PodU64,
+    /// Number of base 10 digits to the right of the decimal place.
+    pub decimals: u8,
+    /// If `true`, this structure has been initialized
+    pub is_initialized: PodBool,
+    /// Optional authority to freeze token accounts.
+    pub freeze_authority: PodCOption<Pubkey>,
+}
+impl IsInitialized for PodMint {
+    fn is_initialized(&self) -> bool {
+        self.is_initialized == PodBool::from_bool(true)
```

Comment:
```suggestion
        self.is_initialized.into()
```
Better?

It would actually be kind of cute if PodBool implemented `PartialEq<bool> for PodBool` so that this line could be:
`self.is_initialized == true`
I can't really find a way to simplify it further to just `self.is_initialized`, though. There's a trait for `Not`, but not one for `Is` aside from PartialEq/Eq.

Reply:
ah yeah, much better! We might need to address the `PartialEq` / `Eq` question in a follow-up PR, but your suggestion is much clearer